app/models/job_queue.rb

```diff
@@ -1,55 +1,48 @@
 # frozen_string_literal: true
+# make jobs with the same queue run in serial and track their status
 class JobQueue
   LOCK = Mutex.new
 
   def initialize
-    @queue = Hash.new { |h, k| h[k] = [] }
+    @queue = Hash.new { |h, q| h[q] = [] }
     @active = {}
-    @registry = {}
   end
 
   def active
     @active.values
   end
 
-  def active?(key, id)
-    @active[key].try(:id) == id
+  def active?(id)
+    active.detect { |je| je.id == id }
   end
 
-  def queued?(key, id)
-    @queue.dup[key].any? { |je| je.id == id }
+  def queued?(id)
+    @queue.values.detect { |jes| jes.detect { |je| return je if je.id == id } }
```

Comment:
It's either having a `queued(id)` and `queued?(id)` method or this non-boolean `?` method ... you can pick ... I don't care much :D